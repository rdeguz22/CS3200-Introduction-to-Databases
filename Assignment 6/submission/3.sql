SELECT
    '1990''s' AS decade,
    g.name AS genre,
    COUNT(m.id) AS movies
FROM genres AS g
LEFT JOIN movies_genres AS mg ON g.name = mg.genre
LEFT JOIN movies AS m
    ON mg.movie_id = m.id
        AND m.year_released >= 1990
    AND m.year_released <= 1999
GROUP BY g.name
UNION
SELECT
    '2000''s' AS decade,
    g.name AS genre,
    COUNT(m.id) AS movies
FROM genres AS g
LEFT JOIN movies_genres AS mg ON g.name = mg.genre
LEFT JOIN movies AS m
    ON mg.movie_id = m.id
        AND m.year_released >= 1990
    AND m.year_released <= 1999
GROUP BY g.name
ORDER BY decade, movies DESC, genre