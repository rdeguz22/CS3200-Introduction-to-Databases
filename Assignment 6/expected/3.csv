DECADE,GENRE,MOVIES
1990's,Drama,9
1990's,Thriller,8
1990's,Crime,5
1990's,Mystery,4
1990's,Action,3
1990's,Sci-Fi,2
1990's,Comedy,1
1990's,Horror,1
1990's,Romance,1
1990's,War,1
1990's,Documentary,0
1990's,Short,0
1990's,Western,0
2000's,Thriller,8
2000's,Comedy,6
2000's,Crime,6
2000's,Drama,6
2000's,Adventure,4
2000's,Action,3
2000's,Fantasy,3
2000's,Romance,3
2000's,Mystery,2
2000's,Sci-Fi,2
2000's,Animation,1
2000's,Family,1
2000's,Horror,1
2000's,Documentary,0
2000's,Short,0
2000's,Western,0
